### PR TITLE
AWS auth method

### PR DESCRIPTION
**Pull request by Anexen**
Implements AWS auth (https://www.vaultproject.io/api-docs/auth/aws).

I'm still working on the tests and I'm going to use [localstack](https://github.com/localstack/localstack) for this.
Vault AWS API has some [deprecated endpoints](https://www.vaultproject.io/api-docs/auth/aws#deprecations-effective-in-vault-1-7). Do you have any suggestions on how to support them (and whether to support them at all)?

### CODE REVIEW

**Comment by jmgilman** [comment]
Thanks for your contribution! There seems to be no reason to include the deprecated endpoints since equivalent functionality exists with the current ones.

**Comment by Anexen** [comment]
A small update. There are two blockers that prevent the successful completion of the tests:
- https://github.com/localstack/localstack/pull/4942
- https://github.com/jmgilman/dockertest-server/pull/8
I will be able to complete the tests when the PRs above are merged and released.

**Comment by jmgilman** [comment]
Thanks for the update. I'll hold off on accepting the `dockertest-server` PR until the `localstack` PR makes it into the next version (so we can bump the container version accordingly).

**Comment by Anexen** [comment]
Hi @jmgilman, I'm done with tests.

**Comment by Anexen** [comment]
Hi @jmgilman, I have finished implementing this feature. Could you please take a look?

File: README.md [hunk 1]
```diff
@@ -21,6 +21,7 @@ The following features are currently supported:
 
 * Auth
   * [AppRole](https://www.vaultproject.io/docs/auth/approle)
+  * [AWS](https://www.vaultproject.io/docs/auth/aws)
```
Comment (jmgilman) [inline README.md:24]: Can you also update the header of `lib.rs` with this line? These two files should stay in sync.

**Comment by jmgilman** [comment]
Thanks for finishing this up, and sorry for the delayed review. It looks great, just needs one minor update to `lib.rs` before merging.

File: README.md (hunk 1 above)
Comment (Anexen) [reply README.md:24]: Updated

**Comment by jmgilman** [comment]
Thanks again!

**Pull request merged**